tests/e2e/connectAutoKeyFynbos.spec.ts

```diff
@@ -0,0 +1,169 @@
+  background,
+  i18n,
+}) => {
+  const username = process.env.FYNBOS_USERNAME!;
+  const password = process.env.FYNBOS_PASSWORD!;
+  const walletAddressUrl = process.env.FYNBOS_WALLET_ADDRESS_URL!;
+
+  test.skip(!username || !password || !walletAddressUrl, 'Missing credentials');
+
+  const { keyId: kid } = await background.evaluate(() => {
+    return chrome.storage.local.get<{ keyId: string }>(['keyId']);
+  });
+
+  const connectButton = await test.step('fill popup', async () => {
+    const connectButton = await fillPopup(popup, i18n, {
+      walletAddressUrl,
+      amount: '10',
+      recurring: false,
+    });
+    return connectButton;
+  });
+
+  await test.step('ensure not logged in', async () => {
+    await page.goto(KEYS_PAGE_URL);
+    expect(page.url()).toBe(LOGIN_PAGE_URL);
+    await page.close();
+  });
+
+  await test.step('ensure key not added already', async () => {
+    const jwksBefore = await getJWKS(walletAddressUrl);
+    expect(jwksBefore.keys.length).toBeGreaterThanOrEqual(0);
+    expect(jwksBefore.keys.find((key) => key.kid === kid)).toBeUndefined();
+  });
+
+  await test.step('asks for key-add consent', async () => {
+    await connectButton.click();
+    await popup.waitForSelector(
+      `[data-testid="connect-wallet-auto-key-consent"]`,
+    );
+
+    expect(popup.getByTestId('connect-wallet-auto-key-consent')).toBeVisible();
+    await popup
+      .getByRole('button', {
+        name: i18n.getMessage('connectWalletKeyService_label_consentAccept'),
+      })
+      .click();
+  });
+
+  page = await test.step('shows login page', async () => {
+    const openedPage = await context.waitForEvent('page', {
+      predicate: (page) => page.url().startsWith(LOGIN_PAGE_URL),
+      timeout: 3 * 1000,
+    });
+    await openedPage.getByLabel('Email').fill(username);
+    await openedPage.getByLabel('Password').fill(password);
+    await openedPage.getByRole('button', { name: 'Log in' }).click();
+    await openedPage.waitForURL(KEYS_PAGE_URL);
+    await expect(openedPage.locator('h1').first()).toHaveText('Keys');
+
+    return openedPage;
+  });
+
+  const continueWaitMsPromise = getContinueWaitTime(context, {
+    walletAddressUrl,
+  });
+
+  const keyNickName = await test.step('adds key to wallet', async () => {
+    const { resolve, promise } = withResolvers<string>();
+    page.on('request', async function interceptApplicationName(req) {
+      if (req.serviceWorker()) return;
+      if (req.method() !== 'POST') return;
+
+      const url = new URL(req.url());
+      if (
+        url.pathname.startsWith('/settings/keys/add-public') &&
+        url.searchParams.get('_data') === 'routes/settings_.keys_.add-public'
+      ) {
+        const applicationName = req.postDataJSON()?.applicationName;
+        resolve(applicationName);
+        page.off('request', interceptApplicationName);
+      }
+    });
+    const keyNickName = await promise;
+    expect(keyNickName).not.toBeFalsy();
```

Comment:
It's intercepting the key add request, to know the key name the code uses. We then use this name to revoke keys with same name.
Intercepting so that: we don't have to copy key name logic here (makes use of browser name); and we can also update the key name during interception if needed.